app/lib/features/tasks/widgets/task_info.dart

```diff
@@ -0,0 +1,409 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:skeletonizer/skeletonizer.dart';
+import 'dart:core';
+
+class TaskInfo extends ConsumerWidget {
+  static const statusBtnNotDone = Key('task-info-status-not-done');
+  static const statusBtnDone = Key('task-info-status-done');
+  static const titleField = Key('task-title');
+  static const dueDateField = Key('task-due-field');
+  static const assignmentsFields = Key('task-assignments');
+  static const selfAssignKey = Key('task-self-assign');
+  static const selfUnassignKey = Key('task-self-unassign');
+  final Task task;
+  const TaskInfo({Key? key, required this.task}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final isDone = task.isDone();
+    final roomId = task.roomIdStr();
+
+    return Column(
+      children: [
+        Card(
+          child: Column(
+            children: [
+              ListTile(
+                leading: InkWell(
+                  key: isDone ? statusBtnDone : statusBtnNotDone,
+                  child: Padding(
+                    padding: const EdgeInsets.only(right: 5),
+                    child: Icon(
+                      isDone
+                          ? Atlas.check_circle_thin
+                          : Icons.radio_button_off_outlined,
+                      size: 48,
+                    ),
+                  ),
+                  onTap: () async {
+                    final updater = task.updateBuilder();
+                    if (!isDone) {
+                      updater.markDone();
+                    } else {
+                      updater.markUndone();
+                    }
+                    await updater.send();
+                  },
+                ),
+                title: TaskTitle(
+                  key: titleField,
+                  task: task,
+                ),
+              ),
+              ListTile(
+                dense: true,
+                leading: const Icon(Atlas.calendar_date_thin),
+                title: Wrap(
+                  children: [
+                    DueChip(
+                      visualDensity: VisualDensity.compact,
+                      key: dueDateField,
+                      canChange: true,
+                      task: task,
+                      noneChild: Text(
+                        'No due date',
+                        style: Theme.of(context).textTheme.bodySmall!.copyWith(
+                              fontWeight: FontWeight.w100,
+                              fontStyle: FontStyle.italic,
+                              color: AppTheme.brandColorScheme.neutral5,
+                            ),
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+              buildAssignees(context, ref),
+              TaskBody(task: task),
```

Comment:
Looks like there are some glitches are there in this feature.
https://github.com/acterglobal/a3/assets/51526480/3601044c-8d2c-47df-8467-90263ba7458f

Reply:
if you don't mind, I'd postpone that preview-bug. Primarily because I am thinking about replacing the entire editor with an inline [Appflowy-Editor](https://pub.dev/packages/appflowy_editor) (see branch `ben-tasks-with-appflowy-editor` that already uses it).